js/completionCalculations.js

```diff
@@ -5,144 +5,152 @@ define([
-                nonAssessmentComponentsCompleted += completionObject.nonAssessmentCompleted;
-                assessmentComponentsTotal += completionObject.assessmentTotal;
-                assessmentComponentsCompleted += completionObject.assessmentCompleted;
-            });
-
-            return {
-                "subProgressCompleted": subProgressCompleted,
-                "subProgressTotal" : subProgressTotal,
-                "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
-                "nonAssessmentTotal": nonAssessmentComponentsTotal,
-                "assessmentCompleted": assessmentComponentsCompleted,
-                "assessmentTotal": assessmentComponentsTotal,
-            };
+        var viewType = contentObjectModel.get('_type');
+        var nonAssessmentComponentsTotal = 0;
+        var nonAssessmentComponentsCompleted = 0;
+        var assessmentComponentsTotal = 0;
+        var assessmentComponentsCompleted = 0;
+        var subProgressCompleted = 0;
+        var subProgressTotal = 0;
+        var isComplete = contentObjectModel.get("_isComplete") ? 1 : 0;
+        var children;
+
+        switch (viewType) {
+            case 'page':
+                // If it's a page
+                var descendantComponents = contentObjectModel.findDescendantModels('components');
+                children = descendantComponents.filter(function(comp) {
+                    return comp.get('_isAvailable') === true && comp.get('_isOptional') === false;
+                });
+
+                var availableChildren = filterAvailableChildren(children);
+                var components = getPageLevelProgressEnabledModels(availableChildren);
+
+                var nonAssessmentComponents = getNonAssessmentComponents(components);
+
+                nonAssessmentComponentsTotal = nonAssessmentComponents.length;
+                nonAssessmentComponentsCompleted = getComponentsCompleted(nonAssessmentComponents).length;
+
+                var assessmentComponents = getAssessmentComponents(components);
+
+                assessmentComponentsTotal = assessmentComponents.length;
+                assessmentComponentsCompleted = getComponentsInteractionCompleted(assessmentComponents).length;
+
+                if (!contentObjectModel.get('_pageLevelProgress')._excludeAssessments) {
+                    subProgressCompleted = contentObjectModel.get("_subProgressComplete") || 0;
+                    subProgressTotal = contentObjectModel.get("_subProgressTotal") || 0;
+                }
+
+                var pageCompletion = {
+                    "subProgressCompleted": subProgressCompleted,
+                    "subProgressTotal": subProgressTotal,
+                    "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
+                    "nonAssessmentTotal": nonAssessmentComponentsTotal,
+                    "assessmentCompleted": assessmentComponentsCompleted,
+                    "assessmentTotal": assessmentComponentsTotal
```

Comment:
Minor: Use of double quotes rather than single.  You could probably get rid of these altogether.